### PR TITLE
Feature/ui

### PR DESCRIPTION
**Pull request by lukebhan**
WIP: Right now just have both the app interface in go and the slide backend communicating. @ayazhafiz Can we possibly discuss what we want to do for arguments right now. I would like it to just be like Wolfram and we apply a bunch of predetermined rules at first. Thoughts?

### CODE REVIEW

**Comment by ayazhafiz** [comment]
> I would like it to just be like Wolfram and we apply a bunch of predetermined rules at first.

I think this is a good idea. We can make it more complicated later on; right now it can just run `./slide <program>`

**Comment by ayazhafiz** [comment]
BTW we should think about versioning too. How will people run different versions of slide/what versions do we want to allow? I am thinking all releases (currently just 0.0.1) and the master branch.

**Comment by lukebhan** [comment]
Eventually, I think our UI will be very thorough. Like possible have a set of default rules you can check and uncheck as well as writing your custom rules. However, I think the average person probably doesn't care what version of slide they are using in the UI, thats much more of a developer problem which I think is why we have the slide-bot/tests and can add it there.

**Comment by lukebhan** [comment]
Also, we probably should just switch from golang to like a react/angular frontend for scalability. Thoughts?

**Comment by ayazhafiz** [comment]
> should just switch from golang to like a react/angular frontend

Go is that backend; that can stay. On the frontend that sounds good, but maybe we should use something like Vue so we can have just a single HTML page.

**Comment by ayazhafiz** [comment]
I see, so currently the view is generated by the server. You can also make a static webpage that will communicate to the server without having to refresh. Lmk if you want to video chat about how to do this, maybe for a live stream on your YT ;)

**Comment by lukebhan** [comment]
Yeah, thats exactly what I mean. I am just unsure how do I run the slide application from a go server? maybe we can chat about this bc I understand setting up the api.

File: app/src/api/main.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+async fn index(info: web::Json<Info>) -> Result<String> {
+    slide(info.input)
+}
+
+#[actix_rt::main]
+async fn main() -> std::io::Result<()> {
+    HttpServer::new(|| App::new().route("/", web::post().to(index)))
+        .bind("127.0.0.1:8088")?.run().await
```
Comment (timothycobrien) [inline app/src/api/main.rs:20]: This should probably be an environment variable

**Comment by lukebhan** [comment]
I think I will just use commands following something like this tutorial now that I explore more.Thanks. 
https://blog.kowalczyk.info/article/wOYk/advanced-command-execution-in-go-with-osexec.html

**Comment by ayazhafiz** [comment]
I recommend keeping the go server, and getting rid of the rust one.

**Comment by lukebhan** [comment]
Frontend pref @ayazhafiz @timothycobrien react or angular are my choices. This probably will be something we all work on so lmk if you have a pref.

**Comment by ayazhafiz** [comment]
I prefer Vue > React > Angular for this project because I would rather keep everything in a single HTML file. However I am not opposed to a larger frontend app with many sources, it is just more maintenance.

**Comment by lukebhan** [comment]
I haven't used vue before, but that is fine. The intro docs look readable so I'm gonna use that. Hoping to have a small little setup that we can expand on.

**Comment by ayazhafiz** [comment]
dope, im very excited for this feature. thanks for doing this

**Comment by ayazhafiz** [comment]
@lukebhan any updates on this?

**Pull request closed without merging**